external/build/youtube-playlist.js

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @fileoverview Fetches the most recent playlists by specified channels from YouTube
+ */
+
+require('dotenv').config();
+
+const fs = require('fs').promises;
+const path = require('path');
+const ms = require('ms');
+const youtube = require('googleapis').google.youtube('v3');
+
+/**
+ * Time interval to determine if the current data set is stale
+ */
+const FETCH_INTERVAL = ms('4h');
+
+/**
+ * Config constants to interact with the YouTube API
+ */
+const API_KEY = process.env.YOUTUBE_API_KEY;
+const PART = ['snippet'];
+const MAX_RESULTS = 50;
+
+/**
+ * Set Google Chrome Developers Youtube channel ID (https://www.youtube.com/@ChromeDevs)
+ */
+const CHANNELS = ['UCnUYZLuoy1rq1aVMwx4aTzw'];
+
+/**
+ * Target directory for the fetched data
+ */
+const targetFile = path.join(__dirname, '../data/youtube-playlist.json');
+const currentTimestamp = Date.now();
+
+const result = {};
+
+/**
+ * Starts the data fetching process form the channel name
+ * specified in the config.
+ */
+
+async function run() {
+  if (!process.env.YOUTUBE_API_KEY) {
+    throw new Error('No `YOUTUBE_API_KEY` environment var for production');
+  }
+
+  /**
+   * Fetch current data set and checks timestamp to see if the data is stale
+   */
+  let dataAge = 0;
+  try {
+    const currentDataRaw = await fs.readFile(targetFile, {encoding: 'utf8'});
+    const currentData = JSON.parse(currentDataRaw);
+
+    dataAge = currentTimestamp - currentData.timestamp;
+  } catch (error) {
+    console.warn(
+      'No existing data found or existing data malformatted. Fetching from YouTube...'
+    );
+  }
+  if (dataAge < FETCH_INTERVAL && dataAge) {
+    console.log(`YouTube data is ${ms(dataAge)} old, no need to fetch.`);
+    return;
+  }
```

Comment:
Sorry I actually changed my mind here. The current behaviour would require us to run `sync-external` before `build-external`, which might not be the best solution. Instead we could do the following: use `fetch` to try to fetch https://www.googleapis.com/storage/v1/b/external-dcc-data/o/youtube-playlist.json (currently 404s) and guesstimate wether a refetch is needed from there.

Look at https://www.googleapis.com/storage/v1/b/external-dcc-data/o/chrome-release.json, there is a `updated` key which is exactly what we want.

Instead of `fetch` try to use @google-cloud/storage, which already is a dependency anyway.

Reply:
I've created a function that using @google-cloud/storage, tries to fetch the data from the bucket, check the timestamp (if exists) and returns a boolean to indicate if the data is stale or not. 

Let me know if that's the direction you had in mind!